common/p_horde.cpp

```diff
@@ -714,7 +716,27 @@ void HordeState::tick()
 		const int offset = P_RandomInt(16) + 30;
 		m_nextPowerup = ::level.time + (offset * TICRATE);
 
-		const mobjtype_t pw = define.randomPowerup().mobj;
+		hordeDefine_t newdef = define;
```

Comment:
If the define has no powerups in the HORDEDEF, the new powerups won't spawn either because this whole section is skipped. It also seems like this could be done without copying the define on every tick. They're unlikely to be very large most of the time but it is copying 4 vectors. Could probably just do it when the define is being selected, since `g_lives` is latched.

Reply:
I fixed that particular bug, but the reason I don't want to modify the defines directly is that it seems the server only loads the defines once, and by injecting them whenever needed, it saves us from having to track and add/remove the powerups depending on cvar changes, and also users might want to put these powerups in their defines and add a level of complexity there.

Reply:
I agree, which is why I said at the time the wave is chosen, not when the defines are loaded.

Reply:
The define has to be modified or horde needs to be restructured to allow temp defines, either doesn't seem very attractive for such a small feature. Also, this runs only on the tic that powerups are spawned in, so there's not much worry about it running every tic

Reply:
Maybe I need to take another look at the larger structure of how this all works. It's been a bit since I did. I thought it would mostly just be taking this and moving it to another spot.
If it works it works though. The edge case of empty powerup defines was the most important to fix.

Reply:
Okay yea I see now why that wouldn't work without modifying the original define.